lib/Node.js

```diff
@@ -27,12 +27,24 @@ class Node {
   }
 
   clear () {
+    if (this.parent) {
+      this._detach()
+    }
+
     for (const child of this.children) {
       child.clear()
     }
 
     this.ptr.deleteOut()
   }
+
+  _detach () {
+    for (const child of this.parent.children) {
+      if (child.term.equals(this.ptr.term)) {
+        this.parent.children.delete(child)
+      }
+    }
```

Comment:
You sure this needs to loop?

Reply:
Sorry, I don't get the question.

One would like to do simply this.parent.children.delete(child), but the term only can be checked for equality using equals(), not ==

Or do you mean to move it into a functional style?

Reply:
Maybe there was some other intent but the code above looks equivalent to

```js
this.parent.children.delete(this)
```

Reply:
You're right is as simple as that.
( I'm always in fear of terms not being equal :) )